pgml-extension/src/orm/snapshot.rs

```diff
@@ -742,26 +737,7 @@ impl Snapshot {
     }
 
     fn select_sql(&self) -> String {
-        format!(
-            "SELECT {} FROM {} {}",
-            self.columns
-                .iter()
-                .map(|c| c.quoted_name())
-                .collect::<Vec<String>>()
-                .join(", "),
-            self.relation_name(),
-            match self.materialized {
-                // If the snapshot is materialized, we already randomized it.
-                true => "",
-                false => {
-                    if self.test_sampling == Sampling::random {
-                        "ORDER BY random()"
-                    } else {
-                        ""
-                    }
-                }
-            },
-        )
+        self.test_sampling.get_sql(&self.relation_name(), self.columns.clone())
```

Comment:
I think this will re-randomize or stratify if the snapshot has been materialized, which would make training runs not reproducible. I think we'll need something akin to the previous version for materialized runs, or this for dynamic runs.

Reply:
ah yes,  you're right. i think i can adjust this to account for whats in `self.materialized `